File: 1 Two Sum/twoSum.java
public class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        Map<Integer, Integer> record = new HashMap<Integer, Integer>();
        for(int i = 0; i < nums.length; i++){
            
            int complement = target - nums[i];
            if( record.containsKey(complement) ){
                int[] res = {i, record.get(complement)};
                return res;
            }
            
            record.put(nums[i], i);
        }

        // Java is really strict on rules. If I don't add these two lines, the compilation cannot pass
        int[] null_res = null;
        return null_res;
    }
}

File: 1 Two Sum/twoSum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        record = {}
        # Note that we have to get the index of each element, so the iteration is slightly different
        for i,element in enumerate(nums):
            complement = target - element
            if complement in record:
                res = [i, record[complement]]
                return res
            record[element] = i